### PR TITLE
Add acceptance test for deleting connetion

### PR DESCRIPTION
**Pull request by terencecho**
## What
Add an acceptance test to check that we are able to delete a connection during a normal state and even when the state of the temporal workflow is in a terminal state. This behavior was originally committed in this PR: https://github.com/airbytehq/airbyte/pull/11302 with unit tests and verified manually, but lacked an acceptance test.

### CODE REVIEW

**Comment by terencecho** [comment]
running into this error in the github builds
```
AcceptanceTests > testActionsWhenTemporalIsInTerminalState() FAILED
    io.grpc.StatusRuntimeException: NOT_FOUND: sql: no rows in result set
        at app//io.grpc.stub.ClientCalls.toStatusRuntimeException(ClientCalls.java:262)
        at app//io.grpc.stub.ClientCalls.getUnchecked(ClientCalls.java:243)
        at app//io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:156)
        at app//io.temporal.api.workflowservice.v1.WorkflowServiceGrpc$WorkflowServiceBlockingStub.terminateWorkflowExecution(WorkflowServiceGrpc.java:2884)
        at app//io.temporal.internal.client.external.GenericWorkflowClientExternalImpl.lambda$terminate$4(GenericWorkflowClientExternalImpl.java:202)
        at app//io.temporal.internal.retryer.GrpcRetryer.lambda$retry$0(GrpcRetryer.java:44)
        at app//io.temporal.internal.retryer.GrpcSyncRetryer.retry(GrpcSyncRetryer.java:61)
        at app//io.temporal.internal.retryer.GrpcRetryer.retryWithResult(GrpcRetryer.java:51)
        at app//io.temporal.internal.retryer.GrpcRetryer.retry(GrpcRetryer.java:41)
        at app//io.temporal.internal.client.external.GenericWorkflowClientExternalImpl.terminate(GenericWorkflowClientExternalImpl.java:195)
        at app//io.temporal.internal.client.RootWorkflowClientInvoker.terminate(RootWorkflowClientInvoker.java:174)
        at app//io.temporal.internal.sync.WorkflowStubImpl.terminate(WorkflowStubImpl.java:376)
        at app//io.airbyte.test.acceptance.AcceptanceTests.testActionsWhenTemporalIsInTerminalState(AcceptanceTests.java:1187)
```
        
 which seems to suggest that it's having problems finding the workflow in the temporal database.

**Comment by terencecho** [comment]
I found it weird that it didn’t work when i tried
```
@EnabledIfEnvironmentVariable(named = "NEW_SCHEDULER",
                              matches = "true")
```
but worked when i did the following
```
    final FeatureFlags featureFlags = new EnvVariableFeatureFlags();
    if (!featureFlags.usesNewScheduler()) {
      LOGGER.info("Skipping test since not using new temporal scheduler");
      return;
    }
```

File: airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/AcceptanceTests.java [hunk 1]
```diff
@@ -1147,6 +1152,57 @@ public void testCancelSyncWhenCancelledWhenWorkerIsNotRunning() throws Exception
     assertEquals(JobStatus.CANCELLED, resp.get().getJob().getStatus());
   }
 
+  @Test
+  @Order(22)
+  @DisabledIfEnvironmentVariable(named = "KUBE",
```
Comment (benmoriceau) [inline airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/AcceptanceTests.java:1157]: Is there a reason to not run on Kube?

**Review by benmoriceau — COMMENTED** [review]
Thanks for improving the acceptance coverage! I only have minor comments.

File: airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/AcceptanceTests.java [hunk 2]
```diff
@@ -1147,6 +1152,57 @@ public void testCancelSyncWhenCancelledWhenWorkerIsNotRunning() throws Exception
...
+    final ConnectionManagerWorkflow connectionManagerWorkflow =
+        workflowCLient.newWorkflowStub(ConnectionManagerWorkflow.class, "connection_manager_" + connectionId);
+    connectionManagerWorkflow.getState();
+
+    // Terminate workflow
+    LOGGER.info("Terminating temporal workflow...");
+    workflowCLient.newUntypedWorkflowStub("connection_manager_" + connectionId).terminate("");
+
+    // expect an exception because the temporal workflow is not running
+    Assertions.assertThatExceptionOfType(ApiException.class)
```
[the remark this answers is not in the document]
Comment (terencecho) [reply airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/AcceptanceTests.java]: makes sense. removed

File: airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/AcceptanceTests.java [hunk 3]
```diff
@@ -1147,6 +1152,57 @@ public void testCancelSyncWhenCancelledWhenWorkerIsNotRunning() throws Exception
...
+    waitForConnectionState(apiClient, connectionId);
+
+    // check if temporal workflow is reachable
+    final ConnectionManagerWorkflow connectionManagerWorkflow =
+        workflowCLient.newWorkflowStub(ConnectionManagerWorkflow.class, "connection_manager_" + connectionId);
+    connectionManagerWorkflow.getState();
+
+    // Terminate workflow
+    LOGGER.info("Terminating temporal workflow...");
+    workflowCLient.newUntypedWorkflowStub("connection_manager_" + connectionId).terminate("");
```
[the remark this answers is not in the document]
Comment (terencecho) [reply airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/AcceptanceTests.java:1220]: done

File: airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/AcceptanceTests.java (hunk 1 above)
Comment (terencecho) [reply airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/AcceptanceTests.java]: originally i thought we would have to do some extra stuff to get it working on kube, but i guess that's not the case. it's been removed.

**Comment by terencecho** [comment]
@benmoriceau PTAL. i refactored it to cover both the normal case and the temporal in a bad state case.

**Review by benmoriceau — APPROVED** [review]

**Pull request merged**